tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/MuleProcessController.java

```diff
@@ -81,6 +81,10 @@ public boolean isDeployed(String appName) {
     return getController().isDeployed(appName);
   }
 
+  public boolean isUndeployed(String appName) {
+    return getController().isUndeployed(appName);
```

Comment:
An application has only two deployment states, deployed or undeployed. So what would be the difference between this and !isDeployed?

Reply:
I completely agree that an app could have only two states; deployed and undeployed. Nevertheless, by the way that isDeployed() API is implemented, I think (also from first-hand experience), that a `true` output of this method totally means that the app got deployed. Nevertheless, a `false` value can't ensure 100% the opposite. I'd say something like that saying of "Correlation does not imply causation"

Reply:
My supporting argument, at least for the un-deployment case, is that relying just on the anchor file it's not enough. There's no a transactional-like behavior on the un-deployment process, and the deletion of the anchor file is just the beginning of that process. I think, there's a high chance that in many cases, as it's happening on Windows, something could happen in the middle of the process, that could leave it halfway. 

Maybe for the deployment scenario, the anchor file gets created at the end of the process, and therefore it's a clear sign that the app got deployed correctly.